File: Assignment/Abstraction/Expressions/BinaryExpressionNode.cs
using Assignment.Extensions;

namespace Assignment.Abstraction.Expressions
{
    internal struct BinaryExpressionNode : IExpressionNode
    {
        public BinaryExpressionNode(INode left, INode right, ArithmeticOperations operation)
        {
            this.Left = left;
            this.Right = right;
            this.Operation = operation;
        }

        public ArithmeticOperations Operation { get; }
        public INode Left { get; }
        public INode Right { get; }

        public string DisplayName()
        {
            return $"{nameof(BinaryExpressionNode)}(operation: {this.Operation.GetDisplayName()})";
        }
    }
}

File: src/main/java/ce305/abstraction/expressions/BinaryExpressionNode.java
package ce305.abstraction.expressions;

import ce305.abstraction.ArithmeticOperation;
import ce305.abstraction.INode;

public class BinaryExpressionNode implements INode {
    public final INode left;
    public final INode right;
    public final ArithmeticOperation operation;

    public BinaryExpressionNode(INode left, INode right, ArithmeticOperation operation) {
        super();
        this.left = left;
        this.right = right;
        this.operation = operation;
    }

    @Override
    public String getDisplayName() {
        return String.format("%(operation: %)", BinaryExpressionNode.class.getName(), this.operation);
    }
}